zlib-rs/src/allocate.rs

```diff
@@ -1,3 +1,5 @@
+#![allow(unpredictable_function_pointer_comparisons)]
```

Comment:
@bjorn3 is this actually risky? We try to be careful in this file about only exposing the function addresses in one way, and miri is OK with the current implementation, but maybe there is still a risk of this misbehaving on some platform?

Reply:
If the comparison incorrectly fails at worst it would be slower. If the comparison succeeds incorrectly, that means the target function either has UB or is functionally equivalent at the assembly level to whichever function we tried to compare against, so it should be fine too.